src/Console/Command/DescribeCommand.php

```diff
@@ -93,21 +129,48 @@ private function describeRectorCodeSamples(array $codeSamples): void
             $codeAfter .= $codeSample->getCodeAfter() . $separator;
         }
 
-        $formattedDiff = $this->differAndFormatter->bareDiffAndFormat($codeBefore, $codeAfter);
-        if ($formattedDiff) {
-            $this->consoleStyle->write($formattedDiff);
+        return [$codeBefore, $codeAfter];
+    }
+
+    private function printWithCliFormat(int $i, InputInterface $input, RectorInterface $rector): void
+    {
+        $this->consoleStyle->section(sprintf('%d) %s', $i, get_class($rector)));
+
+        $rectorDefinition = $rector->getDefinition();
+        if ($rectorDefinition->getDescription()) {
+            $this->consoleStyle->writeln(' * ' . $rectorDefinition->getDescription());
+        }
+
+        if (! $input->getOption(Option::OPTION_NO_DIFFS)) {
+            $this->describeRectorCodeSamples($rectorDefinition->getCodeSamples());
         }
+
+        $this->consoleStyle->newLine(2);
     }
 
-    private function ensureSomeRectorsAreRegistered(): void
+    private function printWithMarkdownFormat(InputInterface $input, RectorInterface $rector): void
```

Comment:
I would decouple `printWithMarkdownFormat` and `printWithCliFormat` from DescribeCommand class. I suggest creating `class MardownFormatter implements Formatter{...}` and `class CliFormatter implements Formatter{...}`.

Reply:
Since it's not needed on multiple places, this would only make maintaining harder.
Not everything that could be abstracted, is good idea to abstract.